gamemode/modules/chat/sv_chat.lua

```diff
@@ -131,9 +131,9 @@ function GM:PlayerSay(ply, text, teamonly) -- We will make the old hooks run AFT
     local DoSayFunc
 
     for k, v in pairs(self.OldChatHooks) do
-        if type(v) ~= "function" then continue end
+        if not isfunction(v) then continue end
 
-        if type(k) == "Entity" or type(k) == "Player" then
+        if isentity(k) or k:IsPlayer() then
```

Comment:
This looks like a potential to error. `k:IsPlayer()` could be run here if `k` is a string, so that call will lead to errors since there isn’t an `IsPlayer` on strings.

The `isentity` check should cover players on its own I believe.

Reply:
Yep, `isentity` returns `true` for a player.

Reply:
Then I will leave the second part as it was.